### PR TITLE
ci: create separate workflow for contract deployment on UAT environment

### PR DESCRIPTION
**Pull request by DeepakBomjan**
* integrate UAT environment running on aws instance to test contract deployment

### Commit Message

```bash
type: commit message
```

see the [guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages) for commit messages.

### Changelog Entry

```bash
version: <log entry>
```

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have documented my code in accordance with the [documentation guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#documentation)
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have run the unit tests
- [x] I only have one commit (if not, squash them into one commit).
- [ ] I have a descriptive commit message that adheres to the [commit message guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages)

> Please review the [CONTRIBUTING.md](/CONTRIBUTING.md) file for detailed contributing guidelines.

### CODE REVIEW

File: .github/workflows/uat-deploy-java-contracts.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+jobs:
+  deploy-contracts-uat:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Deploy all contracts on UAT environment
+        run: cd contracts/javascore && ./gradlew clean build && ./gradlew optimizedJar && ./gradlew deployToUat -PdeploymentENV=uat
```
Comment (nightowl121) [inline .github/workflows/uat-deploy-java-contracts.yml:17]: let's use a different keystore than the godwallet for our uat deployment as it still needs to be authorized.

Comment (nightowl121) [reply .github/workflows/uat-deploy-java-contracts.yml:17]: we can keep the keystore in the git secrets as json file

Comment (DeepakBomjan) [reply .github/workflows/uat-deploy-java-contracts.yml]: Have separated out the keystore and password from coderepo. And have used different keystore for uat deployment.

**Review by nightowl121 — APPROVED** [review]

**Review by redlarva — APPROVED** [review]
looks good for me

**Pull request merged**